Fix twenty-emails not being built during install

A lot of similar reports on Discord:

<img width="718" alt="Screenshot 2024-02-16 at 15 19 11" src="https://github.com/twentyhq/twenty/assets/6399865/77816039-3ee9-4483-a61b-6390d834e33a">
<img width="683" alt="Screenshot 2024-02-16 at 15 19 32" src="https://github.com/twentyhq/twenty/assets/6399865/478d3eac-39f7-4e1c-bbaa-df96f7544587">
<img width="717" alt="Screenshot 2024-02-16 at 15 19 58" src="https://github.com/twentyhq/twenty/assets/6399865/627090df-93ca-4d0f-b3a9-513a502c232f">
<img width="731" alt="Screenshot 2024-02-16 at 15 20 31" src="https://github.com/twentyhq/twenty/assets/6399865/2fa65d4e-9642-4f86-b738-a32b260e38ba">



13 import { PasswordUpdateNotifyEmail } from 'twenty-emails';
                                             ~~~~~~~~~~~~~~~
src/core/auth/services/token.service.ts:22:40 - error TS2307: Cannot find module 'twenty-emails' or its corresponding type declarations.
22 import { PasswordResetLinkEmail } from 'twenty-emails';
                                          ~~~~~~~~~~~~~~~
src/workspace/workspace-cleaner/crons/clean-inactive-workspace.job.ts:9:8 - error TS2307: Cannot find module 'twenty-emails' or its corresponding type declarations.
9 } from 'twenty-emails';

--- a/package.json
+++ b/package.json
@@ -274,6 +274,7 @@
     "ts-jest": "^29.1.1",
     "ts-loader": "^9.2.3",
     "ts-node": "10.9.1",
+    "tsconfig-paths": "^4.2.0",
     "typescript": "^5.3.3",
     "vite": "^5.0.0",
     "vite-plugin-checker": "^0.6.2",

--- a/packages/twenty-server/package.json
+++ b/packages/twenty-server/package.json
@@ -8,7 +8,7 @@
   "scripts": {
     "nx": "NX_DEFAULT_PROJECT=twenty-server node ../../node_modules/nx/bin/nx.js",
     "prebuild": "rimraf dist",
-    "build": "yarn prebuild && nest build --path ./tsconfig.build.json",
+    "build": "npx nx prebuild && nest build --path ./tsconfig.build.json",
     "format": "prettier --write \"src/**/*.ts\" \"test/**/*.ts\"",
     "start": "NODE_ENV=development && nest start",
     "start:dev": "npx nx start --watch",
@@ -20,17 +20,14 @@
     "test:cov": "npx nx test --coverage",
     "test:debug": "node --inspect-brk -r tsconfig-paths/register -r ts-node/register ../../node_modules/.bin/jest --runInBand",
     "test:e2e": "./scripts/run-integration.sh",
-    "typeorm": "npx ts-node -r tsconfig-paths/register ../../node_modules/typeorm/cli.js",
-    "typeorm:migrate": "yarn typeorm migration:run -d ./src/database/typeorm/metadata/metadata.datasource.ts && yarn typeorm migration:run -d ./src/database/typeorm/core/core.datasource.ts",
-    "database:init": "yarn database:setup && yarn database:seed:dev",
-    "database:setup": "npx ts-node ./scripts/setup-db.ts && yarn database:migrate",
-    "database:setup:prod": "npx ts-node ./scripts/setup-db.ts && yarn typeorm:migrate",
+    "typeorm": "npx ts-node ../../node_modules/typeorm/cli.js",
+    "database:init": "npx nx database:setup && npx nx database:seed:dev",
+    "database:setup": "npx ts-node ./scripts/setup-db.ts && npx nx database:migrate",
     "database:truncate": "npx ts-node ./scripts/truncate-db.ts",
-    "database:migrate": "yarn build && yarn typeorm:migrate",
-    "database:seed:dev": "yarn build && yarn command workspace:seed:dev",
-    "database:seed:demo": "yarn build && yarn command workspace:seed:demo",
-    "database:reset": "yarn database:truncate && yarn database:init",
-    "command": "node dist/src/command",
+    "database:migrate": "npx nx typeorm -- migration:run -d src/database/typeorm/metadata/metadata.datasource && npx nx typeorm -- migration:run -d src/database/typeorm/core/core.datasource",
+    "database:seed:dev": "npx nx command -- workspace:seed:dev",
+    "database:seed:demo": "npx nx command -- workspace:seed:demo",
+    "database:reset": "npx nx database:truncate && npx nx database:init",
     "queue:work": "node dist/src/queue-worker"
   },
   "dependencies": {
@@ -39,7 +36,6 @@
     "@ptc-org/nestjs-query-graphql": "patch:@ptc-org/nestjs-query-graphql@4.2.0#./patches/@ptc-org+nestjs-query-graphql+4.2.0.patch",
     "class-validator": "patch:class-validator@0.14.0#./patches/class-validator+0.14.0.patch",
     "graphql-middleware": "^6.1.35",
-    "lodash.isequal": "^4.5.0",
     "passport": "^0.7.0"
   },
   "devDependencies": {

--- a/packages/twenty-server/project.json
+++ b/packages/twenty-server/project.json
@@ -1,4 +1,6 @@
 {
+    "name": "twenty-server",
+    "$schema": "../../node_modules/nx/schemas/project-schema.json",
     "projectType": "application",
     "targets": {
         "build:packageJson": {
@@ -11,6 +13,16 @@
                 "tsConfig": "packages/twenty-server/tsconfig.json",
                 "outputPath": "packages/twenty-server/dist",
                 "updateBuildableProjectDepsInPackageJson": true
+            }
+        },
+        "command": {
+            "executor": "nx:run-commands",
+            "dependsOn": [
+                "build"
+            ],
+            "options": {
+                "cwd": "packages/twenty-server",
+                "command": "node dist/src/command.js"
             }
         },
         "test:debug": {

--- a/packages/twenty-server/scripts/render-run.sh
+++ b/packages/twenty-server/scripts/render-run.sh
@@ -1,4 +1,4 @@
 #!/bin/sh
 export PG_DATABASE_URL=postgres://twenty:twenty@$PG_DATABASE_HOST:$PG_DATABASE_PORT/default
-yarn database:setup:prod
+yarn database:setup
 node dist/src/main

--- a/packages/twenty-server/tsconfig.json
+++ b/packages/twenty-server/tsconfig.json
@@ -27,5 +27,9 @@
       "src/*": ["packages/twenty-server/src/*"],
       "twenty-emails": ["packages/twenty-emails/src/index.ts"]
     }
+  },
+  "ts-node": {
+    "files": true,
+    "require": ["tsconfig-paths/register"]
   }
 }

